File: ProBook/ProBook.Model/Model/NotebookCollection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProBook.Model.Model
{
    public class NotebookCollection
    {
        public int? Id { get; set; }

        public int NotebookId { get; set; }
        public int CollectionId { get; set; }

        public virtual Notebook? Notebook { get; set; } = null!;
        public virtual Collection? Collection { get; set; } = null!;

        public DateTime? CreatedAt { get; set; }
    }
}

File: ProBook/UI/probook-ui/src/app/interfaces/notebook-collection.ts
import { Collection } from "./collection-interface";
import { Notebook } from "./notebook-interface";


export interface NotebookCollection{
    id?:number;
    notebookId?:number;
    collectionId?:number;
    notebook?:Notebook;
    collection?:Collection;
    createdAt?:Date;
}